packages/experiments/src/components/Microfeedback/Microfeedback.base.tsx

```diff
@@ -0,0 +1,161 @@
+/** @jsx withSlots */
+import * as React from 'react';
+import { withSlots } from '../../Foundation';
+import { Stack } from '../../Stack';
+import { classNamesFunction } from '../../Utilities';
+import { IconButton, IButtonStyles, DefaultButton } from 'office-ui-fabric-react/lib/Button';
+import { IStackStyles } from 'office-ui-fabric-react/lib/Stack';
+import { Callout } from 'office-ui-fabric-react/lib/Callout';
+import { FocusZone, FocusZoneDirection } from 'office-ui-fabric-react/lib/FocusZone';
+import { List } from 'office-ui-fabric-react/lib/List';
+import { Text, ITextStyles } from 'office-ui-fabric-react/lib/Text';
+import {
+  IMicrofeedbackProps,
+  IMicrofeedbackViewProps,
+  IMicrofeedbackStyleProps,
+  IMicrofeedbackStyles,
+  VoteType
+} from './Microfeedback.types';
+import { initializeIcons } from '@uifabric/icons';
+
+const getClassNames = classNamesFunction<IMicrofeedbackStyleProps, IMicrofeedbackStyles>();
+
+const microfeedbackStyles: IStackStyles = {
+  root: [
+    {
+      margin: 8,
+      float: 'right'
+    }
+  ]
+};
+
+const microfeedbackItemStyles: IButtonStyles = {
+  root: [
+    {
+      border: '0px',
+      width: '100%'
+    }
+  ]
+};
+
+const microfeedbackTextStyles: ITextStyles = {
+  root: [
+    {
+      margin: '10px'
+    }
+  ]
+};
+
+export interface IMicrofeedbackState extends IMicrofeedbackViewProps {
+  vote: VoteType;
+  isFollowupVisible: boolean;
+}
+
+export class MicrofeedbackBase extends React.Component<IMicrofeedbackProps, IMicrofeedbackState> {
+  // ref's will be linked to each of the icons for callout placement
+  private dislikeRef = React.createRef<HTMLDivElement>();
+  private likeRef = React.createRef<HTMLDivElement>();
+
+  constructor(props: IMicrofeedbackProps) {
+    super(props);
+
+    initializeIcons();
```

Comment:
Components shouldn't be doing this.. this is an app level concern.

Reply:
Fixed